design/ShaderLanguageFAQ.md

```diff
@@ -0,0 +1,157 @@
+
+The W3C WebGPU community group agreed to this direction during the
+February 2020 face-to-face meeting in Redmond.
+
+
+## How do you pronounce “WGSL”?
+
+The W3C WebGPU Community Group does not prescribe a particular
+pronunciation of “WGSL”.
+
+You can say it letter-by-letter, like “W G S L”.
+
+You can say it as a single word, like “wiggzle”, “wiggles”, or “wigsel”.
+
+
+## What does a WGSL shader look like?
+
+WGSL has not yet been specified.
+However, early prototyping work indicates the language can have a look
+and feel as represented in the following examples:
+
+A simple fragment shader:
+
+
+```
+[[location 0]] var<out> gl_FragColor : vec4<f32>;
+
+fn main() -> void {
+    gl_FragColor = vec4<f32>(0.4, 0.4, 0.8, 1.0);
+    return;
+}
+entry_point fragment = main;
+```
+
+
+A code fragment showing if/else:
+
+
+```
+  if (value > 5) {
+    wvar = 14;
+  } elseif (value > 2) {
+    wvar = 12;
+  } else {
+    wvar = 0;
+  }
+```
+
+
+A code fragment showing a switch:
+
+
+```
+switch (a) {
+  case 1: {
+    # The break is implicit
+  }
+  case 2: {
+     fallthrough;
+  }
+  default: {
+  }
+}
+```
+
+
+A code fragment showing a loop:
+
+
+```
+var i : i32 = 0;
+var b : f32 = 2.4;
+loop {
+  break if (i == 4);
+
+  b = b * 3.2;
+  continuing {
+     i = i + 1;
+  }
+}
+```
+
+
+
+## Where can I find the spec?
+
+The WGSL spec will be developed on the WebGPU Community Group area on GitHub.
+See https://github.com/gpuweb/gpuweb/tree/master/wgsl
+
+
+## What browsers support WGSL?
+
+WGSL is the shader language for WebGPU.
+Any browser implementing WebGPU will be required to accept shaders in WGSL.
+Specifically, the WebGPU conformance test suite will use shaders written
+in WGSL.
+
+Browser support for WGSL is expected to be developed over time, along
+with the development of WGSL itself.
+
+
+## Will browsers accept other shader languages for WebGPU?
+
+Accepting a shader language other than WGSL is not part of WebGPU.
+Doing so harms interoperability, as a shader in that other language is
+not portable across browsers.
+
+
+## Where can I find WGSL tooling?
+
+The WebGPU Community Group does not deliver tooling for WGSL.
+However, WebGPU implementers are expected to develop tooling
+for handling WGSL, including parsing, validation, and compilation.
+
+At this time, we expect at least three essentially independent
+implementations of WGSL ingestion as part of browser support for WebGPU.
+Independent implementation serves as an important check on the quality
+of the WGSL and WebGPU specifications.
+
+
+## What about my existing shaders, compilers, and other tooling?
+
+This is also expressed as “the last thing I need is another shader language”.
+(See https://xkcd.com/927/)
+
+For developers invested in the SPIR-V ecosystem for Vulkan and OpenGL,
+the key will be convertibility between WGLSL and SPIR-V.
+For shaders that only use WebGPU features, WebGPU Community Group
+members expect to develop tooling to faithfully convert between WGSL
```

Comment:
Not in the purview of the standardization group, though. I'm not sure what business this sentence has being in the standardization group's repo.

Reply:
The work may not be in the purview, but the question has come up, so having a response to it is good.

Reply:
I'm not disputing having responses to questions. I'm disputing those responses being hosted by the WebGPU standardization group.

Reply:
If the availability of a Forth-to-WGSL was a frequently asked question, I think it would totally deserve a mention in the list of Frequently Asked Questions. As it stands, the toolchain support around WGSL and compatibility with other, existing high-level langauges is probably the most frequently asked question about the project, so I think it makes sense to have an answer for it.